File: WebsiteTinTuc.Admin/aspnet-core/src/WebsiteTinTuc.Admin.Core/Entities/Post.cs
﻿using Abp.Domain.Entities.Auditing;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebsiteTinTuc.Admin.Entities
{
    public class Post : FullAuditedEntity<Guid>
    {
        public string Title { get; set; }
        public string Content { get; set; }
        public string PostUrl { get; set; }
        public long NumberOfViews { get; set; }
        public JobType JobType { get; set; }
        public long MinMoney { get; set; }
        public long MaxMoney { get; set; }
        public MoneyType MoneyType { get; set; }
        public int? TimeExperience { get; set; }
        public DateTime? EndDate { get; set; }
        public ExperienceType ExperienceType { get; set; }
        [ForeignKey(nameof(Company))]
        public Guid CompanyId { get; set; }
        [ForeignKey(nameof(Level))]
        public Guid LevelId { get; set; }
        public virtual Company Company { get; set; }
        public virtual ICollection<CompanyPostHashtag> CompanyPostHashtags { get; set; }
        public virtual Level Level { get; set; }
        public virtual ICollection<CV> CVs { get; set; }
    }

    public enum JobType
    {
        PartTime = 0,
        FullTime = 1
    }

    public enum ExperienceType
    {
        NoExperience = 0,
        Month = 1,
        Year = 2
    }

    public enum MoneyType
    {
        NoNegotiable = 0,
        Negotiable = 1,
    }
}

File: WebsiteTinTuc.Admin/vue/src/store/entities/post.ts
import { JobType } from "../enums/job-type";
import { MoneyType } from "../enums/money-type";
import { ExperienceType } from "../enums/experience-type";
import { HashtagModel } from "./company";
import Entity from "./entity";
import { Guid } from "guid-typescript";

export default class Post extends Entity<Guid> {
    title: string;
    content: string;
    creationTime: Date;
    postUrl: string;
    numberOfViews: number;
    jobType: JobType;
    minMoney: number;
    maxMoney: number;
    moneyType: MoneyType;
    timeExperience?: number;
    endDate?: Date;
    experienceType: ExperienceType;
    companyId: string;
    levelId: string;
    companyName: string;
    levelName: string;
    hashtags: HashtagModel[];
}

export class CategoryHashtagModel extends Entity<Guid> {
    categoryHashtagOfPostId: string;
}